src/JustEat.StatsD/Buffered/BufferExtensions.cs

```diff
@@ -52,9 +52,39 @@ public static bool TryWriteUtf8String(this ref Buffer src, string str)
             return true;
 #endif
         }
+        
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool TryWriteUtf8Chars(this ref Buffer<byte> src, ReadOnlySpan<char> chars)
+        {
+            if (chars.Length == 0)
+            {
+                return true;
+            }
+
+#if NETSTANDARD2_0 || NET461
+            var bytes = Encoding.UTF8.GetBytes(chars.ToArray());
+            return src.TryWrite(bytes);
+#else
+            int written = 0;
+            try
+            {
+                written = Encoding.UTF8.GetBytes(chars, src.Tail);
+            }
+#pragma warning disable CA1031
+            catch (ArgumentException)
```

Comment:
I _think_ having a catch clause in a method breaks the ability to inline this method (maybe I'm thinking of `throw` rather than `catch`). It would be better to test the condition that causes the exception instead if possible, otherwise you can probable delete the attribute from this method and the one above.

Reply:
Yes, you are right. It has no effect.

Reply:
Done!